packages/web/components/icon/icon.tsx

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from "react";
+import { IconProps } from "./types";
+import Star from "./icons/star";
+
+const Icon: React.FC<IconProps> = ({
+  color = "#fff",
+  name,
+  onClick,
+  size = 24,
+}) => {
+  const Child = useMemo(() => {
```

Comment:
My idea was any time a new icon was added, it would be imported and added to the switch case. This frontloads the work as any time an icon is used thereafter, you can just pass the name instead of having to import and render the svg in addition to the Icon component. Can definitely use `children` as well, but thought this was better for usability.